src/pages/home-page.js

```diff
@@ -71,19 +71,63 @@ const HomePage = (props) => {
   const loading = useContext(LoadingContext);
   const { courses } = props;
 
+  // Returns an array of courses sorted in descending order of number of reviews
+  const sortMostReviewed = () => {
+    return Object.values(courses).sort(function(a, b) {
+      return parseInt(b.reviews.length) - parseInt(a.reviews.length);
+    });
+  };
+
+  // This function creates the grid of course review cards
+  const buildGrid = () => {
+    const sortedCourses = sortMostReviewed();
+    const gridArray = [];
+    let gridRow = [];
+    let i = 0;
+
+    // Creates 2 rows of cards, each row being 3 cards long
+    while (i < 2) {
+      for (let i = 0; i < 3; i++) {
+        gridRow.push(sortedCourses.shift());
+      }
+      gridArray.push(gridRow);
+      gridRow = [];
+      i++;
+    }
+    return gridArray.map((row, idx) => {
+      return (
+        <Grid.Row key={idx}>
+          {row.map((eqt) => <Grid.Column key={eqt.id}> <CourseReviewCard
+            code={eqt.courseCode} name={eqt.title} desc={eqt.description}
+            numReviews={eqt.reviews.length}/> </Grid.Column>)}
+        </Grid.Row>
+      );
+    });
+  };
+
+  const [activeTags, setActiveTags] = useState( [] );
```

Comment:
Always declare states at the top of the functional component - put it underneath const { courses } = props line